presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -215,6 +217,8 @@ private static boolean canCastTypeBase(String fromTypeBase, String toTypeBase)
             return true;
         }
         switch (fromTypeBase) {
+            case StandardTypes.INTEGER:
```

Comment:
This sounds correct, but check with @martint that he doesn't see a problem with it